examples/cpp/benchmark/PublisherApp.cpp

```diff
@@ -0,0 +1,545 @@
+    if (subscriber_ == nullptr)
+    {
+        throw std::runtime_error("Subscriber initialization failed");
+    }
+
+    // Create the data reader
+    DataReaderQos reader_qos = DATAREADER_QOS_DEFAULT;
+    subscriber_->get_default_datareader_qos(reader_qos);
+    reader_qos.reliability().kind = config.reliability;
+    reader_qos.durability().kind = config.durability;
+    reader_ = subscriber_->create_datareader(topic_sub_, reader_qos, this, StatusMask::all());
+    if (reader_ == nullptr)
+    {
+        throw std::runtime_error("DataReader initialization failed");
+    }
+}
+
+PublisherApp::~PublisherApp()
+{
+    if (nullptr != participant_)
+    {
+        // Delete DDS entities contained within the DomainParticipant
+        participant_->delete_contained_entities();
+
+        // Delete DomainParticipant
+        DomainParticipantFactory::get_instance()->delete_participant(participant_);
+    }
+}
+
+void PublisherApp::on_publication_matched(
+        DataWriter* /*writer*/,
+        const PublicationMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        startTime = std::chrono::steady_clock::now();
+        std::cout << "Publisher matched." << std::endl;
+        matched_++;
+        cv_.notify_one();
+    }
+    else if (info.current_count_change == -1)
+    {
+        matched_ = static_cast<int16_t>(info.current_count);
+        std::cout << "Publisher unmatched." << std::endl;
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for PublicationMatchedStatus current count change" << std::endl;
+    }
+}
+
+void PublisherApp::on_subscription_matched(
+        DataReader* /*reader*/,
+        const SubscriptionMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        std::cout << "Subscriber matched." << std::endl;
+        matched_++;
+        cv_.notify_one();
+    }
+    else if (info.current_count_change == -1)
+    {
+        std::cout << "Subscriber unmatched." << std::endl;
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for SubscriptionMatchedStatus current count change" << std::endl;
+    }
+}
+
+void PublisherApp::on_data_available(
+        DataReader* reader)
+{
+    SampleInfo info;
+
+    auto actualTime = std::chrono::steady_clock::now();
+    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(actualTime - startTime);
+
+    switch (msg_size_)
+    {
+        case CLIParser::MsgSizeKind::NONE:
+            while ((!is_stopped()) && (RETCODE_OK == reader->take_next_sample(&benchmark_, &info)))
+            {
+                if ((info.instance_state == ALIVE_INSTANCE_STATE) && info.valid_data)
+                {
+                    std::cout << "Sample with index: '" <<
+                        benchmark_.index() << "' (0 Bytes) RECEIVED" << std::endl;
```

Comment:
I found it a little weird to see `0 bytes` when the `unsigned long` of the index is 4 bytes. We may instead writer smth similar to 
```suggestion
                        benchmark_.index() << "' (Array size 0) RECEIVED" << std::endl;
```
This suggestion would apply also to the subscriber